solr/core/src/java/org/apache/solr/core/SolrCores.java

```diff
@@ -193,9 +165,9 @@ List<SolrCore> getCores() {
    * @return An unsorted list. This list is a new copy, it can be modified by the caller (e.g. it
    *     can be sorted).
    */
-  List<String> getLoadedCoreNames() {
+  public List<String> getLoadedCoreNames() {
     synchronized (modifyLock) {
-      return distinctSetsUnion(cores.keySet(), getTransientCacheHandler().getLoadedCoreNames());
+      return new ArrayList<>(cores.keySet());
```

Comment:
would you consider returning an immutable list here? (as in List.copyOf())

Reply:
These methods are documented to return mutable copies, not immutable ones.  If I recall, maybe there were some callers that sorted the result?  I'd prefer immutable... not sure if we should change the contract right now (in 9.x); maybe.